src/appimagetool_fetch_runtime.c

```diff
@@ -86,9 +150,6 @@ bool fetch_runtime(char* arch, size_t* size, char** buffer, bool verbose) {
         fprintf(stderr, "HEAD request to %s failed: %s\n", url, curl_error_buf);
     }
 
-    curl_easy_cleanup(handle);
-    handle = NULL;
```

Comment:
The code below assumes in its error handling that the handle has been cleaned up. If you _really_ want to reuse it, you need to continuously call `curl_easy_cleanup` wherever the function returns.

... this is why we need C++ sooner than later. I consider adding some before the 1.0.0 release. Handling object lifetime correctly is much easier there due to RAII.